src/test/ui/union/union-sized-field.stderr

```diff
@@ -1,26 +1,40 @@
 error[E0277]: the size for values of type `T` cannot be known at compilation time
-  --> $DIR/union-sized-field.rs:4:5
+  --> $DIR/union-sized-field.rs:4:12
    |
 LL | union Foo<T: ?Sized> {
    |           - this type parameter needs to be `std::marker::Sized`
 LL |     value: T,
-   |     ^^^^^^^^ doesn't have a size known at compile-time
+   |            ^ doesn't have a size known at compile-time
    |
-   = help: the trait `std::marker::Sized` is not implemented for `T`
-   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: no field of a union may have a dynamically sized type
+   = help: change the field's type to have a statically known size
+help: borrowed types always have a statically known size
+   |
+LL |     value: &T,
+   |            ^
+help: boxed trait objects have a statically known size
```

Comment:
Isn't this help slightly misleading? `T` is a type parameter, not a trait object here

Reply:
This is a good point. I could say `boxed types have...` and that would be more accurate always at the cost of not giving people some jargon to google... I'll see if I can make it conditional on the type.

Reply:
What about "heap allocated types always have a statically known size"?

Reply:
hmm, I personally don't really like "heap allocated types", as an object of the type is allocated on the heap, not the type itself :sweat_smile:

Not sure what's the best wording here, but maybe "boxed types always have a statically known size" which seems more correct to me

Reply:
I wanted to introduce a bit of jargon here, as `Box` is already mentioned in the code suggestion itself, to tie the concept of "heap allocations" and "`Box`". That being said I see your point. Lets merge this as is to avoid bitrot (I don't think the wording is _outright_ wrong, just suboptimal), but let's open a follow up PR to improve the wording?

Reply:
@lcnr what about "the `Box` type always has a statically known size and allocates its contents in the heap"?